components/notebook-crd/pkg/apis/notebook/v1alpha1/notebook_types.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2019 The Kubeflow Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// NotebookSpec defines the desired state of Notebook
+type NotebookSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+	Template NotebookTemplateSpec `json:"template,omitempty"`
+}
+
+type NotebookTemplateSpec struct {
+	Spec corev1.PodSpec `json:"spec,omitempty"`
```

Comment:
in the original code I prevent certain fields in the PodSpec from being overloaded. This could be a security vulnerability if all the Spec fields could be replaced including the containers